### PR TITLE
#164798196 Deploying Django app to Heroku

### PR DESCRIPTION
**Pull request by dorothyas**
**What does this PR do?**

- Set up and Deploy app to Heroku
- Create pipelines for the application
- Allows access to the base application

**Description of Task to be completed?**

- Add a Procfile at the root directory of the app
- Install the necessary dependencies
- Configure settings.py file for heroku
- Add app.json file to the root directory

**How should this be manually tested?**

The user can access the base application [here](https://ah-backend.herokuapp.com/admin)

**What are the relevant pivotal tracker stories?**

[#164798196](https://www.pivotaltracker.com/n/projects/2320796/stories/164798196)

**Screenshots**

#### Base app

Screenshot 2019-03-27 at 12 01 34

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -142,3 +143,4 @@
     #     'authors.apps.authentication.backends.JWTAuthentication',
     # ),
 }
+django_heroku.settings(locals())
```
Comment (dannylwe) [inline authors/settings.py]: @dorothyas kindly add a new line here.

File: Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web:gunicorn authors.wsgi
```
Comment (dannylwe) [inline Procfile]: @dorothyas kindly add a new line here.

**Review by dannylwe — CHANGES_REQUESTED** [review]
@dorothyas Kindly edit your PR title. It does not follow convention.

File: Pipfile.lock [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "_meta": {
         "hash": {
```
Comment (ja-odur) [inline Pipfile.lock]: @dorothyas Please untrack or delete this file

File: authors/settings.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 import os
 import dj_database_url
 from decouple import config
+import django_heroku
```
Comment (ja-odur) [inline authors/settings.py:16]: @dorothyas Nice work with `django_heroku`

File: Pipfile.lock (hunk 3 above)
Comment (dorothyas) [reply Pipfile.lock]: I added this because the heroku build kept failing without this file

Comment (ja-odur) [reply Pipfile.lock]: okay

**Review by dannylwe — DISMISSED** [review]
Everything looks fine

**Comment by ja-odur** [comment]
@dorothyas Please update the allowed host list in `settings.py` file to include localhosts and the heroku app urls.

**Review by ezrogha — DISMISSED** [review]

**Review by dannylwe — DISMISSED** [review]

File: Pipfile.lock (hunk 3 above)
Comment (dorothyas) [reply Pipfile.lock]: I have deleted it.

**Review by dannylwe — DISMISSED** [review]

**Review by felixkiryowa — DISMISSED** [review]

**Review by ezrogha — DISMISSED** [review]
Well done

**Review by dannylwe — DISMISSED** [review]

**Review by felixkiryowa — APPROVED** [review]

**Review by dannylwe — APPROVED** [review]

**Review by ezrogha — APPROVED** [review]
Well done @dorothyas

File: authors/settings.py (hunk 4 above)
Comment (dorothyas) [reply authors/settings.py:16]: thanks

**Pull request merged**